frame/multi-account/src/lib.rs

```diff
@@ -0,0 +1,1506 @@
+			} else {
+				Err(Error::<T>::MultiAccountNotFound)?
+			}
+			Ok(())
+		}
+
+		/// Cancel a pre-existing, on-going multisig transaction. Any deposit reserved previously
+		/// for this operation will be unreserved on success.
+		///
+		/// The dispatch origin for this call must be _Signed_ and must be the initiator (owner) of the multisig.
+		///
+		/// The multisig transaction is not automatically cancelled if the owner is removed from the multi account,
+		/// nor is it cancelled if the multi account is removed. The multisig transaction can be cancelled by the
+		/// owner even aftre being removed from the multi account, and it can also be cancelled if the multi account
+		/// has been removed.
+		///
+		/// - `multi_account_id`: The account ID of the multi account that was created before.
+		/// - `timepoint`: The timepoint (block number and transaction index) of the first approval
+		/// transaction for this dispatch.
+		/// - `call_hash`: The hash of the call to be executed.
+		///
+		/// # <weight>
+		/// - `O(S)`.
+		/// - Up to one balance-reserve or unreserve operation.
+		/// - One passthrough operation, one insert, both `O(S)` where `S` is the number of
+		///   signatories. `S` is capped by `MaxSignatories`, with weight being proportional.
+		/// - One encode & hash, both of complexity `O(S)`.
+		/// - One event.
+		/// - I/O: 1 read `O(S)`, one remove.
+		/// - Storage: removes one item.
+		/// # </weight>
+		#[weight = SimpleDispatchInfo::FixedNormal(1_000_000)]
+		fn cancel(origin,
+			multi_account_id: T::AccountId,
+			timepoint: Timepoint<T::BlockNumber>,
+			call_hash: [u8; 32],
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			let m = <Multisigs<T>>::get(&multi_account_id, call_hash)
+				.ok_or(Error::<T>::NotFound)?;
+			ensure!(m.when == timepoint, Error::<T>::WrongTimepoint);
+			ensure!(m.depositor == who, Error::<T>::NotOwner);
+
+			let _ = T::Currency::unreserve(&m.depositor, m.deposit);
+			<Multisigs<T>>::remove(&multi_account_id, call_hash);
+
+			Self::deposit_event(RawEvent::MultisigCancelled(who, timepoint, multi_account_id));
+
+			Ok(())
+		}
+	}
+}
+
+impl<T: Trait> Module<T> {
+	/// Derive a multi account ID from the index.
+	pub fn multi_account_id(index: u64) -> T::AccountId {
+		let entropy = (b"modmua/multiacco", index).using_encoded(blake2_256);
+		T::AccountId::decode(&mut &entropy[..]).unwrap_or_default()
+	}
+
+	/// The current `Timepoint`.
+	pub fn timepoint() -> Timepoint<T::BlockNumber> {
+		Timepoint {
+			height: <system::Module<T>>::block_number(),
+			index: <system::Module<T>>::extrinsic_count(),
+		}
+	}
+
+	/// Check that signatories is sorted.
+	fn ensure_sorted(signatories: Vec<T::AccountId>) -> Result<Vec<T::AccountId>, DispatchError> {
+		let mut maybe_last = None;
+		for item in signatories.iter() {
+			if let Some(last) = maybe_last {
+				ensure!(last < item, Error::<T>::SignatoriesOutOfOrder);
+			}
+			maybe_last = Some(item);
+		}
+		Ok(signatories)
+	}
```

Comment:
There is a more idiomatic way to do this:

https://github.com/paritytech/substrate/blob/master/frame/recovery/src/lib.rs#L632

Reply:
Love that, thanks! 👍